docker-compose.yaml

```diff
@@ -97,36 +108,141 @@ services:
     hostname: redis
     restart: always
     command: redis-server --requirepass "${REDIS_AUTH}"
-    ports:
-     - "${REDIS_PORT}:${REDIS_PORT}"
     volumes:
       - "./redis.conf:/usr/local/etc/redis/redis.conf"
     environment:
      - REDIS_REPLICATION_MODE=master
     networks:
       - agr-literature
 
-  agr_literature:
+  api:
     restart: always
-    image: ${REG}/agr_literature_app_test:${TAG}
-    hostname: agr_literature
+    build:
+      context: .
+      dockerfile: docker/api.dockerfile
+      args:
+        REG: ${REG}
+        ALLIANCE_RELEASE: ${ALLIANCE_RELEASE}
+    environment:
+      AWS_SECRET_ACCESS_KEY: "${AWS_SECRET_ACCESS_KEY}"
+      AWS_ACCESS_KEY_ID: "${AWS_ACCESS_KEY_ID}"
+      OKTA_CLIENT_ID: "${OKTA_CLIENT_ID}"
+      OKTA_CLIENT_SECRET: "${OKTA_CLIENT_SECRET}"
+      ENV_STATE: "${ENV_STATE}"
+      PSQL_USERNAME: "${PSQL_USERNAME}"
+      PSQL_PASSWORD: "${PSQL_PASSWORD}"
+      PSQL_HOST: "${PSQL_HOST}"
+      PSQL_PORT: "${PSQL_PORT}"
+      PSQL_DATABASE: "${PSQL_DATABASE}"
+      RESOURCE_DESCRIPTOR_URL: "${RESOURCE_DESCRIPTOR_URL}"
+      HOST: "${HOST}"
+      OKTA_DOMAIN: "${OKTA_DOMAIN}"
+      OKTA_API_AUDIENCE: "${OKTA_API_AUDIENCE}"
+    hostname: literature_api
+    networks:
+      - agr-literature
+    volumes:
+      - agr-logs:/logs
+    depends_on:
+      - postgres
+
+  xml_processing:
+    restart: always
+    build:
+      context: .
+      dockerfile: docker/xml_processing.dockerfile
+      args:
+        REG: ${REG}
+        ALLIANCE_RELEASE: ${ALLIANCE_RELEASE}
+    hostname: literature_xml_processing
     environment:
       API_PORT: "${API_PORT}"
       PYTHONPATH: "/usr/local/bin/src/literature/src/xml_processing:/usr/local/bin/src/literature/backend/app"
-      XML_PATH: "${XML_PATH}"
-      PSQL_DATABASE: "${PSQL_DATABASE}"
```

Comment:
I'm wondering why PSQL_DATABASE is not needed in xml_processing, but is needed in test_runner   Right now the batch sqlalchemy is doing a connection to the database, and that's only in the functional tests in test_runner, but eventually I would think Paulo's refactor would have the xml_processing also use the batch sqlalchemy queries (the code that does it, even for the functional tests, is part of that section of code)

Reply:
I haven't really tested the xml_processing service and I don't know which variables will be needed, but I will add them based on your comment

Reply:
Cool, thanks